cuqi/distribution/_gaussian.py

```diff
@@ -1,41 +1,82 @@
+import warnings
+import numbers
 import numpy as np
+import numpy.linalg as nplinalg
+
 import scipy.stats as sps
-from scipy.sparse import diags, identity, issparse, vstack, csc_matrix, isspmatrix_csr, isspmatrix_csc
-from scipy.sparse import linalg as splinalg
-from scipy.linalg import eigh, eigvals, cholesky
+import scipy.sparse as spa
+import scipy.linalg as splinalg
+
+from cuqi import config
 from cuqi.geometry import _get_identity_geometries
 from cuqi.utilities import force_ndarray, sparse_cholesky
-from cuqi import config
-import warnings
-import numbers
 from cuqi.distribution import Distribution
 
-class GaussianCov(Distribution): # TODO: super general with precisions
+# We potentially allow the use of sksparse.cholmod for sparse Cholesky
+# if it is installed. If not, we use our own sparse Cholesky.
+# TODO: add full support for sparse covariance matrices without cholmod library (missing efficient logdet computation)
+try:
+    import sksparse.cholmod as skchol
+    has_cholmod = True
+except ImportError:
+    has_cholmod = False
+
+
+class Gaussian(Distribution):
     """
-    General Gaussian probability distribution. Generates instance of cuqi.distribution.GaussianCov
+    General Gaussian probability distribution. Generates instance of cuqi.distribution.Gaussian.
+
+    Can be defined by a mean and one of the following: covariance, precision, square root of covariance, or square root of precision.
+
+    Internally the class will always convert the given matrices to the square root of the precision matrix.
```

Comment:
This is a general thing for distributions, which I think should be made clear somewhere else. I hope after my other changes its not required to understand what it is to use the Gaussian, I added an issue though #128

Reply:
My main issue with this is if the user is given an error message talking about the "Gaussian with mutable parameters", I don't think it will be clear to users why we don't simply say "Gaussian", at that point it is not like we have other Gaussian to distinguish from, so do we need in error messages to say "Gaussian with mutable parameters".